src/dsm/headers/Agent.hpp

```diff
@@ -17,47 +17,45 @@
 #include "../utility/Typedef.hpp"
 
 #include <concepts>
-#include <stdexcept>
 #include <limits>
 #include <optional>
+#include <stdexcept>
+#include <vector>
 
 namespace dsm {
   /// @brief The Agent class represents an agent in the network.
-  /// @tparam Id, The type of the agent's id. It must be an unsigned integral type.
-  /// @tparam Size, The type of the size of a street. It must be an unsigned integral type.
   /// @tparam delay_t, The type of the agent's delay. It must be a numeric type (see utility/TypeTraits/is_numeric.hpp).
   template <typename delay_t>
     requires(is_numeric_v<delay_t>)
   class Agent {
   private:
     Id m_id;
-    Id m_itineraryId;
+    std::vector<Id> m_trip;
     std::optional<Id> m_streetId;
     std::optional<Id> m_srcNodeId;
     delay_t m_delay;
     double m_speed;
     double m_distance;    // Travelled distance
     unsigned int m_time;  // Travelled time
+    size_t m_itineraryIdx;
 
   public:
     /// @brief Construct a new Agent object
     /// @param id The agent's id
     /// @param itineraryId The agent's itinerary
-    Agent(Id id, Id itineraryId);
+    /// @param srcNodeId Optional, The id of the source node of the agent
+    Agent(Id id, Id itineraryId, std::optional<Id> srcNodeId = std::nullopt);
     /// @brief Construct a new Agent object
     /// @param id The agent's id
-    /// @param itineraryId The agent's itinerary
-    /// @param srcNodeId The id of the source node of the agent
-    Agent(Id id, Id itineraryId, Id srcNodeId);
+    /// @param itineraryIds The agent's itinerary
+    /// @param srcNodeId Optional, The id of the source node of the agent
+    Agent(Id id, std::vector<Id> const& trip, std::optional<Id> srcNodeId = std::nullopt);
     /// @brief Set the street occupied by the agent
     /// @param streetId The id of the street currently occupied by the agent
     void setStreetId(Id streetId) { m_streetId = streetId; }
-    /// @brief Set the source node id of the agent
-    /// @param srcNodeId The id of the source node of the agent
-    void setSourceNodeId(Id srcNodeId) { m_srcNodeId = srcNodeId; }
     /// @brief Set the agent's itinerary
     /// @param itineraryId The agent's itinerary
-    void setItineraryId(Id itineraryId) { m_itineraryId = itineraryId; }
+    void setItineraryId(std::vector<Id> itineraryId) { m_trip = itineraryId; }
```

Comment:
Move

Reply:
Actually, this function is unused beacuse we have the constructor... I'm going to remove it